library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java

```diff
@@ -1237,12 +1237,29 @@ private void seekToInternal(SeekPosition seekPosition) throws ExoPlaybackExcepti
                 /* forceBufferingState= */ playbackInfo.playbackState == Player.STATE_ENDED);
         seekPositionAdjusted |= periodPositionUs != newPeriodPositionUs;
         periodPositionUs = newPeriodPositionUs;
-        updatePlaybackSpeedSettingsForNewPeriod(
-            /* newTimeline= */ playbackInfo.timeline,
-            /* newPeriodId= */ periodId,
-            /* oldTimeline= */ playbackInfo.timeline,
-            /* oldPeriodId= */ playbackInfo.periodId,
-            /* positionForTargetOffsetOverrideUs= */ requestedContentPositionUs);
+        if (periodId.equals(playbackInfo.periodId)) {
+          if (shouldUseLivePlaybackSpeedControl(playbackInfo.timeline, periodId)) {
+            int windowIndex = playbackInfo.timeline.getPeriodByUid(periodId.periodUid, period).windowIndex;
+            playbackInfo.timeline.getWindow(windowIndex, window);
+            if (requestedContentPositionUs != C.TIME_UNSET) {
+              livePlaybackSpeedControl.setTargetLiveOffsetOverrideUs(
+                  getLiveOffsetUs(playbackInfo.timeline, periodId.periodUid, requestedContentPositionUs));
+            } else {
+              if (!playbackInfo.timeline.isEmpty()) {
+                // Seek resolved to window.getDefaultPositionMs(), so just remove the override.
+                livePlaybackSpeedControl.setTargetLiveOffsetOverrideUs(C.TIME_UNSET);
+              }
+            }
+          }
```

Comment:
basically this is an 'inline' of the logic from `updatePlaybackSpeedSettingsForNewPeriod` when the periodId does not change.

I'm curious for the source update call to `updatePlaybackSpeedSettingsForNewPeriod`, I'm assuming this is the possible in a DASH playlist to have multiple live periods?

Reply:
Thanks for the contribution!

`updatePlaybackSpeedSettingsForNewPeriod` is called each time when the playing period potentially changed. This is not necessarily a multi-period stream. Currently this is called from `seekToInternal` (after a seek), `handleMediaSourceListInfoRefreshed` (after the timeline has changed), and `maybeUpdateReadingPeriod` when the reading period advances to the next period. 

From this I rather think we should move this logic into `updatePlaybackSpeedSettingsForNewPeriod`. So we can call it in any case because we want to potentially reset the playback parameters in the `!shouldUseLivePlaybackSpeedControl(...)` case.

The only issue with this is that `updatePlaybackSpeedSettingsForNewPeriod` is potentially called from the other call sites with `C.TIME_UNSET`. When `C.TIME_UNSET` is passed `updatePlaybackSpeedSettingsForNewPeriod` currently only remove the override when the old period is different to the new period (for instance when moving to the next reading period or when the timeline change has removed the playing period).

Concretely, I think your change does actually what `updatePlaybackSpeedSettingsForNewPeriod` does except that L1945 [1] does only remove the override when the periods are not the same, what is not the case in the `seekTo` case.

Allow me to suggest the following so you can test whether this is equivalent for you:

- add a parameter `boolean positionRequestedBySeek` to the parameter list of `updatePlaybackSpeedSettingsForNewPeriod`
- All call sites call the method then with `false` except the call from `seekToInternal`.
- Revert your change in `seekToInternal` and instead call `updatePlaybackSpeedSettingsForNewPeriod` with `positionRequestedBySeek=true`
- change ExoPlayerImplInternal.java#L1945 to `if (positionRequestedBySeek || !Util.areEqual(oldWindowUid, windowUid))`

I think that should boil down to the same, avoid some duplicated code and makes sure that the `!shouldUseLivePlaybackSpeedControl(..)` case is covered as well.

Can you apply this and probably even more important, test this with your app/stream to see if it is equivalent to your change? 

[1] https://github.com/google/ExoPlayer/blob/release-v2/library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java#L1945

Reply:
> Thanks for the contribution!
You're welcome

> `updatePlaybackSpeedSettingsForNewPeriod` is called each time when the playing period potentially changed.

Ah, ok as long as this alway the case then yes, it would duplicate less code to keep it all in that method.  Guess we need a name like `updatePlaybackSpeedSettingsForNewPeriodOrSeek` (:-)) 

> 
> From this I rather think we should move this logic into `updatePlaybackSpeedSettingsForNewPeriod`. So we can call it in any case because we want to potentially reset the playback parameters in the `!shouldUseLivePlaybackSpeedControl(...)` case.

That sounds good, I'll rework the code to follow this pattern and submit a second commit. 

> 
> The only issue with this is that `updatePlaybackSpeedSettingsForNewPeriod` is potentially called from the other call sites with `C.TIME_UNSET`. When `C.TIME_UNSET` is passed `updatePlaybackSpeedSettingsForNewPeriod` currently only remove the override when the old period is different to the new period (for instance when moving to the next reading period or when the timeline change has removed the playing period).
> 
> Concretely, I think your change does actually what `updatePlaybackSpeedSettingsForNewPeriod` does except that L1945 [1] does only remove the override when the periods are not the same, what is not the case in the `seekTo` case.

Correct, the main reason I pull the code out was to make it clear what is going on, and to avoid adding a parameter to 
`updatePlaybackSpeedSettingsForNewPeriod`.  If we are good for adding the parameter it will remove the duplicated logic.  
> Allow me to suggest the following so you can test whether this is equivalent for you:
> ...
> I think that should boil down to the same, avoid some duplicated code and makes sure that the `!shouldUseLivePlaybackSpeedControl(..)` case is covered as well.
> 
> Can you apply this and probably even more important, test this with your app/stream to see if it is equivalent to your change?

Sure, happy to do that. Also, you want me to cherry-pick the commit to `androidx.media` or does that just make managing the two repositories more of a headache for your team?

Reply:
SG, thanks! It's fine to keep it here for this PR that is a single file. But you are right, for future PRs we'd prefer `androidx.media` if that's possible.